src/status_im2/common/toasts/events.cljs

```diff
@@ -0,0 +1,50 @@
+(ns status-im2.common.toasts.events
+  (:require
+   [status-im.utils.fx :as fx]
+   [taoensso.encore    :as enc]))
+
+(def ^:private next-toast-id (atom 0))
+
+(fx/defn upsert
+  {:events [:toasts/upsert]}
+  [{:keys [db]} id opts]
+  (let [{:toasts/keys [index toasts]} db
+        update?                       (some #{id} index)
```

Comment:
could you pls elaborate on (some #{id} index)

Reply:
To check if the upsert call is an update or insert and use this to maintain the ordered (db :toasts/index)